types/artifacts.d.ts

```diff
@@ -429,6 +429,8 @@ declare global {
         isCss: boolean;
         /** Flags whether this element was contained within a <picture> tag. */
         isPicture: boolean;
+        /** Flags whether this element was contained within a ShadowRoot */
+        isShadow: boolean;
```

Comment:
🚲 🏠  WDYT about `isInShadowDOM`?

a suggestion because there are many relationships I think of with `isShadow`

- it could be an element that *has* a `shadowRoot`  but is still in light DOM
- it could be a shadow root itself (what `isShadow` alone makes me think of)
- it could be a child element underneath a shadow root in the shadow DOM (our actual situation here)

Reply:
thanks for explaining some of the caveats of what the presence of a `shadowRoot` can mean! 
for the 1st bullet point, if an element has a `shadowRoot`, but shows up in the light DOM, do you know if the `getNodePath` function should still break? / should I worry about this edge case?
I'll switch it to isInShadowDOM because that encapsulates exactly what we're looking for!

Reply:
> if an element has a shadowRoot, but shows up in the light DOM, do you know if the getNodePath function should still break

I don't think we ever target such a node right now. I don't know much about custom elements with shadow DOM but my understanding is that's basically the "shadow host" [in this explanation](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM) which is the custom element wrapper, and we're only ever targeting the elements *inside* it

![image](https://user-images.githubusercontent.com/2301202/88978102-aab92d80-d284-11ea-908c-c2f4ce128335.png)

Long way of saying, the third-option is the only situation I'm aware of that could ever apply to an image element, so I totally understand your choice of simplifying to `isShadow` but I still think clarity wins out for anyone reading this in the future who is still learning how all the shadow stuff works :)

Reply:
sounds good! thanks!